components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx

```diff
@@ -247,12 +243,18 @@ export const PortfolioAsset = (props: Props) => {
       }
       const foundToken = tokensWithCoingeckoId?.find(token => token.coingeckoId.toLowerCase() === coinMarket?.id?.toLowerCase())
       setIsTokenSupported(foundToken !== undefined)
-    } else {
-      setIsTokenSupported(true)
+      return token
     }
-
-    return token
-  }, [assetId, userVisibleTokensInfo, selectedTimeline, tokenId])
+    if (!contractOrSymbol || !chainIdOrSymbol) {
+      return undefined
+    }
+    // If the id length is greater than 15 assumes it's a contractAddress
+    const userToken = contractOrSymbol.length > 15
+      ? userVisibleTokensInfo.find((token) => token.contractAddress === contractOrSymbol && token.chainId === chainIdOrSymbol)
+      : userVisibleTokensInfo.find((token) => token.symbol.toLowerCase() === contractOrSymbol.toLowerCase() && token.chainId === chainIdOrSymbol && token.contractAddress === '')
```

Comment:
I think we can get rid of the `> 15` condition now, and replace this block with something like this:

```
const userToken = userVisibleTokensInfo.find(token =>
  (token.contractAddress.toLowerCase() === contractOrSymbol.toLowerCase() && token.chainId === chainIdOrSymbol) ||
  (token.symbol.toLowerCase() === contractOrSymbol.toLowerCase() && token.chainId === chainIdOrSymbol && token.contractAddress === ''))
```

Reply:
Updated this, also added an additional check for `tokenId` that I forgot to include for NFT's.